feat(app): App advanced settings u2e info

koji opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview
Move USB-to-Ethernet adapter information from Robot settings to AppSettings Advancedsettings.
Display the adapter information such as Description(deviceName), Manufacturer, and Driver version(for Windows).

This fixes #9849

Design
https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=1297%3A87906

<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Changelog
- Added a new section to Advanced Settings
- Added test cases for U2E adapter information to Addvanced Settings test
<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests

<!--
Describe any requests for your reviewers here.
-->

# Risk assessment
low
<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

koji committed:
Added U2B information section

koji committed:
Added device info to the U2E information section

koji committed:
Added two text to settings.json

koji committed:
Finish styling

koji committed:
Fix link styling

koji committed:
remove test code for styling

koji committed:
Update AdvancedSettings.tsx

koji committed:
Added tests for U2E adapter info

koji committed:
Update AdvancedSettings.test.tsx

koji committed:
added a new test case for the coverage

koji committed:
Updated UI for not connected case

koji committed:
remove console.log

koji committed:
Added the test case for not connected USB-to-Ethernet

b-cooper commented on app/src/organisms/AppSettings/AdvancedSettings.tsx:
```diff
@@ -223,6 +239,99 @@ export function AdvancedSettings(): JSX.Element {
+                  <StyledText as="p" color={COLORS.darkBlack}>
+                    {t('usb_to_ethernet_adapter_toast_message')}
+                  </StyledText>
+                  <Link
```
For these links that are external to the app, we have a component at `app/src/atoms/Link/ExternalLink` that already captures these styles. You should be able to use it here.

koji replied:
@b-cooper Actually I used `Link` on purpose since Emily told me that we would need to keep the link color dark black and no icon because the link is inside a warning message.
Seems `ExternalLink` doesn't have a property to turn off the icon.  Do you have any good idea for this link?

b-cooper replied:
Ok sounds good 👍 . Thanks for the explanation.

b-cooper reviewed (approved):
This addition looks good!  


I noticed that the background color of the page is slightly different (grey/white) right around where this setting appears.  <img width="94" alt="Screen Shot 2022-04-07 at 5 54 14 PM" src="https://user-images.githubusercontent.com/4731953/162326183-9532802c-83d8-422f-9d59-74c9e298e234.png">

This is definitely unrelated to your changes, but it looks like making one of the top level divs in the `NextGenApp.tsx` file have `min-height: 100%` instead of `height: 100%` fixes the issue. 

It might be worth making that change in this PR since it is so contained.  


```
      <Box width="100%">
        <Breadcrumbs pathCrumbs={pathCrumbs} />
        <Box
          position={POSITION_RELATIVE}
          width="100%"
          height="100%" // <--- change this to minHeight
          backgroundColor={COLORS.background}
          overflow={OVERFLOW_SCROLL}
        >
          <ModalPortalRoot />
```

koji committed:
sync with edge
Merge branch 'edge' into app_advanced_settings_u2e_info

koji commented:
> This addition looks good!
> 
> I noticed that the background color of the page is slightly different (grey/white) right around where this setting appears. <img alt="Screen Shot 2022-04-07 at 5 54 14 PM" width="94" src="https://user-images.githubusercontent.com/4731953/162326183-9532802c-83d8-422f-9d59-74c9e298e234.png">
> 
> This is definitely unrelated to your changes, but it looks like making one of the top level divs in the `NextGenApp.tsx` file have `min-height: 100%` instead of `height: 100%` fixes the issue.
> 
> It might be worth making that change in this PR since it is so contained.
> 
> ```
>       <Box width="100%">
>         <Breadcrumbs pathCrumbs={pathCrumbs} />
>         <Box
>           position={POSITION_RELATIVE}
>           width="100%"
>           height="100%" // <--- change this to minHeight
>           backgroundColor={COLORS.background}
>           overflow={OVERFLOW_SCROLL}
>         >
>           <ModalPortalRoot />
> ```

I though `minHeight` made sense, but actually, it fixed the entire screen and didn't allow me to scroll for items under U2E information. We should keep height="100%", shouldn't we?

koji commented:
@b-cooper I noticed the background thing wouldn't come from `NextGenApp.tsx` but `AppSettings.tsx`
The current AppSettings is a little bit different from Figma's design.

koji committed:
Add external prop

koji committed:
Modified height to align the design

koji committed:
Sync with this branch to modify the layout
Merge branch 'app_app_settings_layout' into app_advanced_settings_u2e_info

koji committed:
fixed typo

koji commented:
@b-cooper I updated a few files to modify AppSettings. (added padding and fixed the issue you mentioned in this PR).
Could you take a look again?
Also, I noticed a couple of typos in labware.
